### PR TITLE
Correctly determine case IDs in xforms

### PR DESCRIPTION
**Pull request by kaapstorm**
This addresses [FB 178655](http://manage.dimagi.com/default.asp?178655) and [FB 178686](http://manage.dimagi.com/default.asp?178686).

@snopoke, cc @emord @gcapalbo

### CODE REVIEW

File: corehq/apps/app_manager/util.py [hunk 1]
```diff
@@ -545,3 +545,18 @@ def change_unique_id(form):
                 jsonpath_update(reference, id_changes[reference.value])
 
     return app_source
+
+
+def append_update(dict_, new_dict):
```
Comment (snopoke) [inline corehq/apps/app_manager/util.py]: Is this really a more generally useful function or is it just here for the tests?

File: corehq/apps/app_manager/tests/test_submodule_xform.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    FormActionCondition,
+    LoadUpdateAction,
+    Module,
+    OpenSubCaseAction,
+    UpdateCaseAction,
+)
+from corehq.apps.app_manager.tests import TestFileMixin
+
+
+class SubmoduleXFormBase(TestFileMixin):
```
Comment (snopoke) [inline corehq/apps/app_manager/tests/test_submodule_xform.py]: These should all go in the `test_child_module.py` file (which should perhaps be renamed to `test_submodule.py`)

File: corehq/apps/app_manager/tests/test_submodule_xform.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    file_path = ('data', 'submodule_xform')
+    child_module_class = None
+
+    def setUp(self):
```
Comment (snopoke) [inline corehq/apps/app_manager/tests/test_submodule_xform.py]: These tests might be less code if you use `AppFactory`: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/tests/app_factory.py#L6-6

(would be good to get your feedback on it too)

File: corehq/apps/app_manager/util.py (hunk 1 above)
Comment (kaapstorm) [inline corehq/apps/app_manager/util.py]: Yeah, it's here for the test. I couldn't run the test when it was nested, and it felt wrong in suite_xml.py when it wasn't nested. The test isn't amazingly useful, so I'm happy to put it back.

**Comment by kaapstorm** [comment]
Left append_update as it was, and moved tests.

**Comment by snopoke** [comment]
Still need a test for this use case: https://github.com/dimagi/commcare-hq/pull/7944

File: corehq/apps/app_manager/tests/test_child_module.py [hunk 4]
```diff
@@ -1,7 +1,18 @@
+from corehq.apps.app_manager.tests import AppFactory
 from django.test import SimpleTestCase
 from corehq.apps.app_manager.const import APP_V2
-from corehq.apps.app_manager.models import FormActionCondition, OpenSubCaseAction, UpdateCaseAction, ParentSelect, \
-    PreloadAction, Module, LoadUpdateAction, AdvancedModule, Application
+from corehq.apps.app_manager.models import (
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_child_module.py]: F401 'OpenCaseAction' imported but unused

**Pull request merged**